lib/Persistence/TrafficLimiter.php

```diff
@@ -99,6 +123,32 @@ public static function canPass()
         if (self::$_limit < 1) {
             return true;
         }
+    
+        // Check if $_ipKey is exempted from ratelimiting
+        if (!is_null(self::$_exemptedIp)) {
+            $exIp_array = explode(',', self::$_exemptedIp);
+            foreach ($exIp_array as $ipRange) {
+                // Match $_ipKey to $ipRange and if it matches it will return with a true
+                $address = \IPLib\Factory::addressFromString($_SERVER[self::$_ipKey]);
+                $range   = \IPLib\Factory::rangeFromString(trim($ipRange));
+                // If $range is null something went wrong (possible invalid ip given in config)
+                if ($range == null) {
+                    $contained = false;
```

Comment:
Hmm instead of using this helper variable  $contained`, just always return `false` immediately, and only if it goes through completely, return `true`.

Reply:
Yes I did that on purpose, as i'm looping thru a list of ipranges. And let's say iprange2 is not valid but the other ranges are, it should not return false. And should goto next iprange. But that can be changed if wanted.

Reply:
Oh indeed you are correct. Of course it needs to do that in this case. Though one could outsource the check for each one iprange into a private function, so then you would just loop through the ranges and call the private function and check it’s return value. As soon as that is `true`, you could exclude it from rate limiting.

That would be a possibility to refactor it a little nicer, I’d suggest.